core/modules/czar/python/appInterface.py

```diff
@@ -108,21 +125,25 @@ def submitQuery(self, query, conditions):
             return ("error", "error",
                     "error locking result, check qserv/db config.")
         context = app.Context(conditions)
-        a = app.InbandQueryAction(query, context,
+        a = app.InbandQueryAction(userquery, context,
                                   lock.setSessionId, resultName)
         if a.getIsValid():
             self._maybeCallWithThread(a.invoke)
             lock.unlockAfter(self._threadFunc, a.getResult)
         else:
             lock.unlock()
-            return ("error","error",a.getError())
+            error = (a.getError(), False, False, False)
+            logger.err("Returning error to proxy: {0}".format(error))
+            return error
 
         # Remember client context for kill-operations
         proxyName = conditions["client_dst_name"]
         proxyThread = conditions["server_thread_id"]
         self._clientToServerId[(proxyThread, proxyName)] = a.sessionId
 
-        return (resultName, lockName, "")
+        result = (False, resultName, lockName, a.proxyOrderBy)
```

Comment:
This seems to contradict your description, `error` is the first element of the tuple, documentation says it must be last.